frontend/lib/resolution/scope-queries.cpp

```diff
@@ -486,17 +505,49 @@ static bool doLookupInScope(Context* context,
     // Search parent scopes, if any, until a module is encountered
     const Scope* cur = nullptr;
     bool reachedModule = false;
+    bool skipClosestConditional = false;
+    bool didSkipConditional = false;
+
+    // This trickiness is required to implement correct scoping behavior
+    // for 'if-vars' in conditionals. The 'if-var' lives in the scope
+    // for the conditional, but it is not visible within the 'else'
+    // branch. Without this hack, we'd be able to see the 'if-var' in
+    // both branches. First, detect if the start scope is the else-block.
+    if (!scope->id().isEmpty())
+      if (auto ast = parsing::idToAst(context, scope->id()))
+        if (isElseBlockOfConditionalWithIfVar(context, ast))
```

Comment:
Recommend you make this into a parsing query that takes in the ID (so we can cache the result of running it). I think it'd be reasonable to start that query by calling `idToTag`. Using this approach will allow the code here to have more incremental reuse (since the uAST pointers will change relatively often).